### PR TITLE
Feature/check tmp files

### LINKED ISSUE #107: Check if `tmp` files are deleted
We are creating tmp files when we create a new file for ex or when e rename or move one. In those cases tmp files are create in tmp OS dependent folder (`/tmp` on Linux for ex.).
We need to check if those get deleted after operation completes or at least when the process finishes.

Test case
- start the cli
- create several files
- rename
- write content
- delete some, but not all
- check those there is no data in `/tmp` of in OS specific tmp folder

### PR DESCRIPTION
**Pull request by srzamfir**
This one covers some basic test cases against the whole lib
* the common part spins the whole rencfs lib and uses `/tmp/rencfs/data` and `/tmp/rencfs/mnt` as data and mount directories
* then all tests are executed against the mount folder (`/tmp/rencfs/mnt`)
* each test creates its own file/folder structure and cleans up in the end 
* if any test panics then unmount is executed automatically by the Drop implementation of the common part

Some limitations of this approach:
* the async nature of the underlying fs operations makes testing specifics in `/tmp/rencfs/data/inodes` for example really difficult (and with unpredictable results)
* for example if in a test you'd create some files (on the mount path) and then check that the number of files in `/tmp/rencfs/data/inodes` or `/tmp/rencfs/data/contents` corresponds to your expected number of files => you will generate random results (some temporary inodes and actual files are still available when you do the check so the numbers will not match, sometimes the test will luckily pass)
* as a result most of tests focus on testing specifics , like the existence of a specific inode
* deleting test cases become more interesting as the `fs::remove_*` functions don't necessarily guarantee the files are immediately removed (especially if the fh are still open)

### CODE REVIEW

File: tests/rencfs_itest.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
+mod common;
```
Comment (radumarias) [inline tests/rencfs_itest.rs]: maybe have suggestive name for this file

File: tests/rencfs_itest.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        let bytes_written = fh.unwrap().write_all("test".as_bytes());
+        assert!(
+            bytes_written.is_ok(),
+            "failed to write [{}]",
+            bytes_written.err().unwrap()
+        );
+    }
+    // warning! remove does not guarantee immediate removal so this leaks inodes
+    let res = fs::remove_file(path);
+    assert!(res.is_ok(), "failed to delete [{}]", res.err().unwrap());
```
Comment (radumarias) [inline tests/rencfs_itest.rs]: shouldn't we check that tmp was deleted? or what is the purpose of these tests?

**Review by radumarias — CHANGES_REQUESTED** [review]

File: tests/rencfs_itest.rs (hunk 1 above)
Comment (srzamfir) [reply tests/rencfs_itest.rs]: done .. this is now mount_setup

File: tests/rencfs_itest.rs (hunk 2 above)
Comment (srzamfir) [reply tests/rencfs_itest.rs]: the test setup assumes that we mount in /tmp/rencfs/mnt where we interact programmatically with various fs functions : e.g. we create a file, write some contents, rename it, check its attributes , add a directory and so on; at the end of each test we clean up 
the temp check test actually runs the suggested scenario and compares the before and after contents of /tmp => the results seem to show we don't leak anything there

**Comment by radumarias** [comment]
I see it's still draft, let me know when it's ready for review, but looks ok to me

**Comment by srzamfir** [comment]
it seems the build is still broken ...  not sure what happens there ; i'm guessing it has something to do with the config

**Comment by radumarias** [comment]
something related to this branch?

**Comment by srzamfir** [comment]
The build still fails on the AUR step but i believe that's already another topic

**Review by radumarias — APPROVED** [review]

**Pull request merged**